system/jlib/jtrace.hpp

```diff
@@ -55,6 +66,33 @@ class jlib_decl LogTrace
     void setLocalId(const char* id);
 };
 
+interface ISpan : extends IInterface
+{
+    virtual void setSpanAttribute(const char * key, const char * val) = 0;
+    virtual void setSpanAttributes(const IProperties * attributes) = 0;
+    virtual void addSpanEvent(const char * eventName) = 0;
+    virtual void setActive() = 0;
```

Comment:
I debated not exposing this because it is very openTel specific, but ultimately decided the caller should have the ability to set a particular trace as the active (parent span for all new spans). If we don't expose this, or similar functionality the caller will be responsible for tracking and explicitly setting the parent span which is sensible I suppose